src/base/bittorrent/session.cpp

```diff
@@ -1658,17 +1658,12 @@ const QStringList Session::getListeningIPs()
     Logger* const logger = Logger::instance();
     QStringList IPs;
 
-    //Take the override addresss
-    const QString networkAddr = pref->getNetworkAddress();
-    if ( !networkAddr.isEmpty()) {
-        IPs.append( networkAddr);
-        return IPs;
-    }
-
     const QString ifaceName = pref->getNetworkInterface();
+    const QString ifaceAddr = pref->getNetworkInterfaceAddress();
     const bool listenIPv6 = pref->getListenIPv6();
 
-    if (ifaceName.isEmpty()) {
+    //No interface name or address defined just use an empty list
+    if (ifaceName.isEmpty() && ifaceAddr.isEmpty()) {
```

Comment:
This isn't correct. The ifaceName might be empty but the ifaceAddr could be not. Thus the test for valid interface name(a few lines below) will fail and result in binding to 127.0.0.1.
So maybe this is better:
```c++
if (ifaceName.isEmpty()) {
    if (!ifaceAddr.isEmpty())
        IPs.append(ifaceAddr);
    else
        IPs.append(QString());

    return IPs;
}
```